src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -1484,6 +1493,85 @@ bool QgsLayoutItemLegend::isRefreshing() const
 }
 
 
+void QgsLayoutItemLegend::syncLayersWithUpdatedCanvasMinimumMaximum()
+{
+
+  if ( mCustomLayerTree )
+    return;
+
+  if ( !mMap )
+    return;
+
+  QgsLegendModel *model = this->model();
+  if ( !model )
+    return;
+
+  QgsLayerTreeGroup *root = model->rootGroup();
+  if ( !root )
+    return;
+
+  const QList<QgsLayerTreeLayer *> layers = root->findLayers();
+
+  for ( QgsLayerTreeLayer *layerTreeLayer : layers )
+  {
+    QgsMapLayer *mapLayer = layerTreeLayer->layer();
+
+    double min = std::numeric_limits<double>::quiet_NaN();
+    double max = std::numeric_limits<double>::quiet_NaN();
+    bool found = false;
+
+    if ( QgsRasterLayer *rasterLayer = qobject_cast<QgsRasterLayer *>( mapLayer ) )
+    {
+      QgsRasterRenderer *renderer = rasterLayer->renderer();
+
+      QgsRasterMinMaxOrigin minMaxOrigin = renderer->minMaxOrigin();
+
+      if ( minMaxOrigin.extent() == Qgis::RasterRangeExtent::UpdatedCanvas )
+      {
+        rasterLayer->computeMinMax( renderer->inputBand(), minMaxOrigin, minMaxOrigin.limits(),
+                                    mMap->extent(), static_cast<int>( QgsRasterLayer::SAMPLE_SIZE ),
+                                    min, max );
+
+        found = !( std::isnan( min ) && std::isnan( max ) );
+      }
+    }
+
+    if ( QgsMeshLayer *meshLayer = qobject_cast<QgsMeshLayer *>( mapLayer ) )
+    {
+      QgsRenderContext renderContext = QgsLayoutUtils::createRenderContextForMap( mMap, nullptr );
```

Comment:
This is an expensive operation, we should only do it once, outside of the layer loop